app/javascript/elm/src/Main.elm

```diff
@@ -403,32 +407,36 @@ update msg model =
             in
             ( { model | overlay = Overlay.update overlay model.overlay }, Cmd.none )
 
-        ToggleIndoor ->
+        ToggleIndoor newValue ->
             let
                 ( subModel, subCmd ) =
                     deselectSession model
             in
-            if subModel.isIndoor then
-                ( { subModel | isIndoor = False, overlay = Overlay.update (RemoveOverlay IndoorOverlay) model.overlay }
-                , Cmd.batch [ Ports.toggleIndoor False, subCmd ]
-                )
+            if subModel.isIndoor == newValue then
+                ( subModel, Cmd.none )
 
-            else
+            else if newValue then
                 ( { subModel | isIndoor = True, profiles = LabelsInput.empty, overlay = Overlay.update (AddOverlay IndoorOverlay) model.overlay }
                 , Cmd.batch [ Ports.toggleIndoor True, Ports.updateProfiles [], subCmd ]
                 )
 
-        ToggleStatus ->
+            else
+                ( { subModel | isIndoor = False, overlay = Overlay.update (RemoveOverlay IndoorOverlay) model.overlay }
+                , Cmd.batch [ Ports.toggleIndoor False, subCmd ]
+                )
+
+        ToggleStatus newStatus ->
             let
                 ( subModel, subCmd ) =
                     deselectSession model
-
-                newStatus =
-                    Status.toggle model.status
             in
-            ( { subModel | status = newStatus }
-            , Cmd.batch [ Ports.toggleActive (Status.toBool newStatus), subCmd ]
-            )
+            if subModel.status == newStatus then
+                ( subModel, Cmd.none )
```

Comment:
subCmd here as well

Reply:
fyi after talking offline went with returning the model instead